### PR TITLE
Fix DDR vTable Indexing

### PR DESCRIPTION
**Pull request by fengxue-IS**
- Add algorithm version for vtable
- update vtable access methods to use J9VTableHeader structure

Close: #2488 

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
@gacholio Can you take a loop please

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9VTablesCommand.java [hunk 1]
```diff
@@ -52,45 +55,58 @@ public void run(String command, String[] args, Context context, final PrintStrea
...
+			 * 		Skip both and start from the third element in vTable which is the first virtual method reference.
+			 */
+			for (long i = startIndex; i < vTableMethodCount.longValue(); i++) {
+				String name = J9MethodHelper.getName(J9MethodPointer.cast(vTable.at(i)));		    	
+				String intAddr = U8Pointer.cast(vTable.at(i)).getHexAddress();
+				if (!jitVTable.isNull()) {		            
+					String jitAddr = U8Pointer.cast(jitVTable.at(vTableMethodCount.sub(i))).getHexAddress();
+					CommandUtils.dbgPrint(out, String.format(" %d\t!j9method %s\t%s\t%s\n", i, intAddr, jitAddr, name));
+				} else {
+					CommandUtils.dbgPrint(out, String.format(" %d\t!j9method %s\t%s\t\n", i, intAddr, name));
```
Comment (fengxue-IS) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9VTablesCommand.java]: this change will make ddr vtable command `<!j9vtables>` with old core files to start display at index 2, and new cores start at index 0 (currently starts from 1)

File: runtime/ddr/algorithm_versions.c [hunk 2]
```diff
@@ -38,6 +38,7 @@
 #define VM_STACK_GROW_VERSION 1
 #define ALG_ROM_HELP_VERSION 1
 #define FOUR_BYTE_OFFSETS_VERSION 1
+#define ALG_VM_VTABLE_VERSION 1
```
[the remark this answers is not in the document]
Comment (fengxue-IS) [reply runtime/ddr/algorithm_versions.c]: fixed

**Comment by gacholio** [comment]
jenkins test all xlinux jdk8

**Pull request merged**